pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -6,13 +6,15 @@
 from pytensor.graph.fg import FunctionGraph
 from pytensor.link.utils import fgraph_to_python
 from pytensor.raise_op import CheckAndRaise
-from pytensor.tensor.basic import Alloc, AllocEmpty, ARange, Eye, Join
+from pytensor.tensor.basic import Alloc, AllocEmpty, ARange, Eye, Join, MakeVector
 
 
 @singledispatch
 def pytorch_typify(data, dtype=None, **kwargs):
     r"""Convert instances of PyTensor `Type`\s to PyTorch types."""
-    return torch.as_tensor(data, dtype=dtype)
+    if data is not None:
+        return torch.as_tensor(data, dtype=dtype)
+    return None
```

Comment:
We should dispatch on `NoneType`:

```python
@pytorch_typify.register(NoneType):
def pytorch_typify_None(data, **kwargs):
  return None
```

Reply:
* Where should this go?
* In addition to condition in `def pytorch_typify(data, dtype=None, **kwargs): ...` ?

Reply:
It should be fine in dispatch.basic.py.

If you dispatch you don't need the if, because the dispatch mechanism already chooses which function to call based on the type of data

Reply:
I tried something like:

```python
from pytensor.tensor.type_other import NoneTypeT
@pytorch_typify.register(NoneTypeT)
def pytorch_typify_None(data, **kwargs):
    return None
```
but the condition is still required. I checked the JAX backend for reference and there is also a similar condition there for `pytorch_typify(data ...`
    

```

Reply:
A different issue surfaced. I don't know whether it is related to these changes or not. 

Calling `repeat` with `axis=None` results in an error in `ElementWise` at [this point](https://github.com/pymc-devs/pytensor/blob/c6d85d1bfb03e5ff5f5c1ac288293af13a5c5396/pytensor/tensor/elemwise.py#L792); namely:

> torch._dynamo.exc.InternalTorchDynamoError: 'int' object has no attribute 'shape'

the values of `inputs` is `(tensor(17), 3, 2)`. The corresponding graph is shown below.

```
Reshape{1} [id A] <Vector(float64, shape=(?,))> 8
 ├─ Alloc [id B] <Matrix(float64, shape=(?, 3))> 7
 │  ├─ ExpandDims{axis=1} [id C] <Matrix(float64, shape=(?, 1))> 6
 │  │  └─ Reshape{1} [id D] <Vector(float64, shape=(?,))> 5
 │  │     ├─ a [id E] <Matrix(float64, shape=(?, ?))>
 │  │     └─ [-1] [id F] <Vector(int64, shape=(1,))>
 │  ├─ Mul [id G] <Scalar(int64, shape=())> 4
 │  │  ├─ Shape_i{0} [id H] <Scalar(int64, shape=())> 1
 │  │  │  └─ a [id E] <Matrix(float64, shape=(?, ?))>
 │  │  └─ Shape_i{1} [id I] <Scalar(int64, shape=())> 0
 │  │     └─ a [id E] <Matrix(float64, shape=(?, ?))>
 │  └─ 3 [id J] <Scalar(int64, shape=())>
 └─ MakeVector{dtype='int64'} [id K] <Vector(int64, shape=(1,))> 3
    └─ Mul [id L] <Scalar(int64, shape=())> 2
       ├─ 3 [id J] <Scalar(int64, shape=())>
       ├─ Shape_i{0} [id H] <Scalar(int64, shape=())> 1
       │  └─ ···
       └─ Shape_i{1} [id I] <Scalar(int64, shape=())> 0
          └─ ···

```

Reply:
Typify should be registered on NoneType (python) not NoneTypeT (PyTensor)

Reply:
Regarding you error, there shouldn't be integers 3 and 2 as inputs, it should be tensor(2) and tensor(3).

We need to track where those come from and fix it

Reply:
Probably Shape_i is not correctly implemented and is returning integers?